_posts/2021-05-07-spring-cloud-aws-rds.md

```diff
@@ -0,0 +1,363 @@
+
+```java
+@Configuration
+public class ApplicationConfiguration {
+  @Bean
+  public RdsInstanceConfigurer instanceConfigurer() {
+    return ()-> {
+        TomcatJdbcDataSourceFactory dataSourceFactory = 
+                      new TomcatJdbcDataSourceFactory();
+        dataSourceFactory.setInitialSize(10);
+        dataSourceFactory.setValidationQuery("SELECT 1 FROM DUAL");
+        return dataSourceFactory;
+    };
+  }
+}
+```
+
+Here we are overriding the validation query and the initial size during instantiation of `dataSourceFactory`.
+
+## Injecting the Data Source
+This data source can now be injected into any Spring Bean like our repository class in our example as shown here: 
+
+```java
+@Service
+public class SystemRepository {
+  
+   private final JdbcTemplate jdbcTemplate;
+
+   @Autowired
+   public SystemRepository(DataSource dataSource) {
+        this.jdbcTemplate = new JdbcTemplate(dataSource);
+   }
+  
+  public String getCurrentDate() {
+    String result = jdbcTemplate.queryForObject(
+      "SELECT CURRENT_DATE FROM DUAL", new RowMapper<String>(){
+
+      @Override
+      public String mapRow(ResultSet rs, int rowNum) 
+          throws SQLException {
+              return rs.getString(1);
+      }     
+    });
+    return result;
+  }
+
+}
+
+```
+As we can see here, it is completely decoupled from the database configuration. We can easily change the database configuration or the database itself( to MySQL or PostgreSQL, or Oracle) in RDS without any change to the code. 
+
+If we work with multiple data source configurations inside one application context, we need to qualify the data source injection point with an @Qualifier annotation.
+
+## Running the Example
+With our data source set up and injected in a repository class, let us now run this example with a JUnit test:
+
+```java
+@SpringBootTest
+class SpringcloudrdsApplicationTests {
+  
+  @Autowired
+  private SystemRepository systemRepository;
+
+  @Test
+  void testCurrentDate() {
+    String currentDate = systemRepository.getCurrentDate();
+    System.out.println("currentDate "+currentDate);
+  }
+}
+
+```
+Once again, this is nothing specific to Spring Cloud here. All the magic happens in the configuration. 
+
+In this JUnit test, we are invoking our repository class method to print the current date. The output log after running the test is shown below:
+```shell
+ :: Spring Boot ::                (v2.4.5)
+
+... : Starting SpringcloudrdsApplicationTests using Java 14.0.1 
+...
+...
+Loading class `com.mysql.jdbc.Driver'. This is deprecated. \
+The new driver class is `com.mysql.cj.jdbc.Driver'...
+currentDate 2021-05-12
+... : Shutting down ExecutorService 'applicationTaskExecutor'
+
+```
+We can see a warning in the log for using a deprecated driver class which is safe to be ignored. We have not specified any driver class here. The driver class `com.mysql.jdbc.Driver` is registered based on the metadata read from the database connection to AWS RDS. 
```

Comment:
Is there a ticket somewhere in the Spring Cloud AWS project that tracks this deprecation?

Reply:
It is deprecated by MySQL connector/J api. Spring cloud aws jdbc is still referring to the earlier name (checked the source code)

The name of the class that implements java.sql.Driver in MySQL Connector/J has changed from com.mysql.jdbc.Driver to com.mysql.cj.jdbc.Driver. The old class name has been deprecated.